### ISSUE TITLE
data/packet_operating_system: Incorrect error handling

### ISSUE THREAD

**Issue by radeksimko** [terraform-providers/terraform-provider-packet#155]
### Terraform Version

```
Terraform v0.12.0
+ provider.packet v2.2.1
```

### Terraform Configuration Files
```hcl
data "packet_operating_system" "esxi" {
  name             = "VMware ESXi"
  distro           = "vmware"
  version          = "BLABLAH"
  provisionable_on = "c1.xlarge.x86"
}
```

### Expected Behavior

```
data.packet_operating_system.esxi: Refreshing state...

Error: There are no operating systems that match the search criteria

  on main.tf line 1, in data "packet_operating_system" "esxi":
   1: data "packet_operating_system" "esxi" {

```

### Actual Behavior

```
data.packet_operating_system.esxi: Refreshing state...

Error: There is more than one operating system that matches the search criteria

  on main.tf line 1, in data "packet_operating_system" "esxi":
   1: data "packet_operating_system" "esxi" {

```

### Steps to Reproduce

1. `terraform apply`

---

I'm not sure if this is an API bug or provider bug - I haven't done any debugging at all, but it looked confusing enough for me to at least file it as an issue.

I assume it would be helpful if the "more than one" error message also contained more details about what was returned from the API - e.g. how many systems and which ones; with some reasonable limits obviously - we don't want to return all available OS in case of too loose match. I reckon <= 5-10 would suffice for some basic debugging?

**Comment by t0mk** [comment]
@radeksimko thanks I will take a look

**Comment by t0mk** [comment]
@radeksimko I fixed this in #156. If you have time, please review the linked PR.
